### PR TITLE
Support parsing `DROP INMEMORY JOIN GROUP` in Oracle

### PR DESCRIPTION
**Pull request by chakkk309**
Ref https://github.com/apache/shardingsphere/issues/6480.

- Add `DROP INMEMORY JOIN GROUP` grammar
- Support parsing `DROP INMEMORY JOIN GROUP` in oracle
- Add test

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@chakkk309 Please solve code conflict first.

**Comment by chakkk309** [comment]
> @chakkk309 Please solve code conflict first.

Done.

**Comment by chakkk309** [comment]
> @chakkk309 Please solve code conflict first.

Done.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java [hunk 1]
```diff
@@ -1404,16 +1405,19 @@ public final class SQLParserTestCases {
...
     private final List<DropRollbackSegmentStatementTestCase> dropRollbackSegmentTestCases = new LinkedList<>();
-    
+
     @XmlElement(name = "create-lockdown-profile")
     private final List<CreateLockdownProfileStatementTestCase> createLockdownProfileTestCases = new LinkedList<>();
-    
+
     @XmlElement(name = "drop-lockdown-profile")
     private final List<DropLockdownProfileStatementTestCase> dropLockdownProfileTestCases = new LinkedList<>();
-
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java:1413]: Please keep same indent with previous line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java (hunk 1 above)
Comment (chakkk309) [reply shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java]: Done.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**